patches/amd-mainline/llvm-project/0001-Enable-MSVC-support-in-comgr.patch

```diff
@@ -0,0 +1,30 @@
+From 20b027cca8ce5d1cc816d2ecbf2150ce811cc3f0 Mon Sep 17 00:00:00 2001
+From: Scott <scott.todd0@gmail.com>
+Date: Fri, 14 Feb 2025 15:22:48 -0800
+Subject: [PATCH] Enable MSVC support in comgr.
+
+---
+ amd/comgr/CMakeLists.txt | 7 ++++++-
+ 1 file changed, 6 insertions(+), 1 deletion(-)
+
+diff --git a/amd/comgr/CMakeLists.txt b/amd/comgr/CMakeLists.txt
+index ed01d21cea30..e643256f2a89 100644
+--- a/amd/comgr/CMakeLists.txt
++++ b/amd/comgr/CMakeLists.txt
+@@ -228,7 +228,12 @@ elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+     "/wd4624" #[[Suppress 'derived class' : destructor could not be generated because a base class destructor is inaccessible]]
+     "/wd4267" #[[Suppress 'var' : conversion from 'size_t' to 'type', possible loss of data]]
+     "/wd4291" #[[Suppress 'declaration' : no matching operator delete found; memory will not be freed if initialization throws an exception]]
+-    "/wd4146" #[[Suppress 'unary minus operator applied to unsigned type, result still unsigned]])
++    "/wd4146" #[[Suppress 'unary minus operator applied to unsigned type, result still unsigned]]
++
++    # Enable standards-conforming preprocessor.
++    # https://learn.microsoft.com/en-us/cpp/build/reference/zc-preprocessor
++    "/Zc:preprocessor"
```

Comment:
Clues that helped settle on this fix:

* https://github.com/llvm/llvm-project/blob/60af83506a3aa379c59e0f9793ce7815d726aee1/llvm/cmake/modules/HandleLLVMOptions.cmake#L594-L598
* https://github.com/ROCm/llvm-project/issues/38 (this had the same error messages that I saw when building locally)

Should upstream the patch somehow.

Reply:
Yeah, ping the LLVM team next week. Thanks for the sleuthing.